cinema/views.py

```diff
@@ -2,7 +2,9 @@
 
 from django.db.models import F, Count
 from rest_framework import viewsets
+from rest_framework.authentication import TokenAuthentication
 from rest_framework.pagination import PageNumberPagination
+from cinema.permissions import IsAdminOrIfAuthenticatedOrReadOnly
```

Comment:
Ensure that the `IsAdminOrIfAuthenticatedOrReadOnly` permission class is correctly implemented in `permissions.py`. There was an issue in the `permissions.py` file that needs to be fixed for this import to work correctly.